sdk/trace/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessor.java

```diff
@@ -128,26 +133,34 @@ public CompletableResultCode forceFlush() {
     private final long scheduleDelayNanos;
     private final int maxExportBatchSize;
     private final long exporterTimeoutNanos;
-
     private long nextExportTime;
-
-    private final BlockingQueue<ReadableSpan> queue;
-
+    private final ConcurrentLinkedQueue<ReadableSpan> queue;
+    private final int maxQueueSize;
+    private final AtomicInteger queueSize = new AtomicInteger(0);
+    private final AtomicLong addedSpansCounter = new AtomicLong(0);
     private final AtomicReference<CompletableResultCode> flushRequested = new AtomicReference<>();
     private volatile boolean continueWork = true;
-    private final ArrayList<SpanData> batch;
+    private final Collection<SpanData> batch;
```

Comment:
why change the type here? any particular reason?

Reply:
This is a stray change from rebasing. Not necessary.